jobserver/forms.py

```diff
@@ -114,12 +113,18 @@ def __init__(self, project, repos_with_branches, *args, **kwargs):
 
         # has there been a repo selected already?
         if "data" in kwargs and "repo" in kwargs["data"]:
-            repo = first(
-                self.repos_with_branches,
-                key=lambda r: r["url"] == kwargs["data"]["repo"],
+            repo = next(  # pragma: no branch
+                (
+                    r
+                    for r in self.repos_with_branches
+                    if r["url"] == kwargs["data"]["repo"]
+                ),
+                None,
```

Comment:
thought: `None` is not a valid value for the subsequent lines, which check attributes of the value. We could replace these defaults with a `StopException` which should never be reached and warns as such. 

Note that we don't do any validation of inputs in the [code above](https://github.com/opensafely-core/job-server/pull/4503/files#diff-8a8820228bf2d257ba3f9c3c703d55f1247e7be2a2d8d5f7ff61352d19103739R105-R108). Therefore, I'm not sure whether to start adding such validation here.